### PR TITLE
DE-3841: Convert to poetry

### PR DESCRIPTION
**Pull request by VladimirShteyn**
## JIRA ticket
[](https://tetrascience.atlassian.net/browse/DE-3841)

## Description
* Moved content from Pipfile and setup.py into pyproject.toml. Turns out `typing-extensions` wasn't used anywhere in `ts-ids-validator` so it could be removed as a package dependency.
* One small word change of `pipenv` -> `poetry` in the README

## Do we want to release this as 0.9.12?
 - [ ] Run `poetry version 0.9.12` before merge.

## Automated testing

- [ ] Unit tests
- [ ] Integration tests
- [ ] API tests
- [ ] End-To-End tests
- [ ] N/A (please clarify)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-  - pip install pipenv
-  - pipenv install --dev
+  - pip install --upgrade pip setuptools wheel
+  - pip install poetry
+  - poetry install
 script:
-  - pipenv run python -m pytest --cov=ids_validator --cov-branch --cov-report xml
+  - pytest --cov=ids_validator --cov-branch --cov-report xml
   - sed -i "s;$(pwd);/usr/src;g" coverage.xml
   - docker run -v "$(pwd):/usr/src" sonarsource/sonar-scanner-cli -Dsonar.projectKey=ts-ids-validator -Dsonar.login="$SONAR_TOKEN" -Dsonar.host.url="$SONAR_HOST"
```
Comment (abclay) [inline .travis.yml]: This isn't in-scope for this ticket, but I bet @aslotte would like to know about this remaining Sonar integration.

**Review by abclay — APPROVED** [review]
Quick work!

**Review by spinwang — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (aslotte) [reply .travis.yml]: Nice catch @abclay! Please remove.

Comment (mlipschultz) [reply .travis.yml]: That probably also explains why the build is failing in Travis: https://app.travis-ci.com/github/tetrascience/ts-ids-validator/builds/250134781

**Review by mlipschultz — APPROVED** [review]
All changes look good.

Can you remove the build status from the top of the README?  It's referencing travis, which we're not using anymore.  This should have been caught in the PR where we actually transitioned away from travis, but I noticed it now.

**Pull request merged**